src/slime_parser.peg.eex

```diff
@@ -1,32 +1,33 @@
-document <- doctype? tags_list crlf* eof;
+document <- doctype? tags? eof;
 
-doctype <- space? 'doctype' space (!eol .)+ eol;
+doctype <- space? 'doctype' space name:(!eol .)+ eol;
 
-% TODO: We can parse nested structure like this:
-% tags_list <- tag (crlf tag)*;
-% tag <- verbatim_text / space? tag_item (crlf empty_lines:(crlf space?)* indent tags_list dedent)? / '' &.;
-% Also remove dedent from eol definition and modify transforms accordingly
-tags_list <- tag (crlf indent? empty_lines:(space? crlf indent?)* tag:tag dedent*)*;
+tag <- comment / verbatim_text / tag_item;
+tag_item <- space? (embedded_engine / inline_html / code / html_tag);
 
-% NOTE: Leading whitespace is handled separately for comments and verbatim text
-tag <- comment / verbatim_text / space? tag_item / blank:(space? &eol);
+tags <- (tag crlf*)+;
+nested_tags <- crlf+ indent tags dedent;
 
-tag_item <- (embedded_engine / inline_html / code / html_tag);
+inline_html <- &'<' text_content nested_tags?;
```

Comment:
Slim does not check for the closing tag. It's an interesting idea — it could be handled similar to `if-else` — but I think it is out of scope of this PR, yes.